test/submission.js

```diff
@@ -2816,13 +2814,11 @@ module.exports = function(app, template, hook) {
             var submission = helper.getLastSubmission();
             assert.equal(helper.lastResponse.statusCode, 400);
             assert.equal(helper.lastResponse.body.name, 'ValidationError');
-            assert.equal(helper.lastResponse.body.details.length, 3);
-            assert.equal(helper.lastResponse.body.details[0].message, 'Text Field must be a non-empty array');
+            assert.equal(helper.lastResponse.body.details.length, 2);
+            assert.equal(helper.lastResponse.body.details[0].message, 'Text Field must be an array');
```

Comment:
It seems to me that this way of validating array data models is fine, because rather than contain both "this is not an array" and "this is not a non-empty array" errors is redundant. In the core validation rules, if the data model should be an array and isn't, it will return the former. If it should be an array and is empty (and required), it will return the latter.